src/VisualStudio/Xaml/Impl/Implementation/LanguageServer/Handler/Diagnostics/WorkspacePullDiagnosticHandler.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Editor.Xaml;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServer;
+using Microsoft.CodeAnalysis.LanguageServer.Handler;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.LanguageServices.Xaml.Features.Diagnostics;
+using Microsoft.VisualStudio.LanguageServices.Xaml.Implementation.LanguageServer.Extensions;
+
+namespace Microsoft.VisualStudio.LanguageServices.Xaml.Implementation.LanguageServer.Handler.Diagnostics
+{
+    [ExportLspMethod(MSLSPMethods.WorkspacePullDiagnosticName, mutatesSolutionState: false, StringConstants.XamlLanguageName), Shared]
+    internal class WorkspacePullDiagnosticHandler : AbstractPullDiagnosticHandler<WorkspaceDocumentDiagnosticsParams, WorkspaceDiagnosticReport>
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public WorkspacePullDiagnosticHandler(
+            ILspSolutionProvider solutionProvider,
+            IXamlPullDiagnosticService xamlPullDiagnosticService)
+            : base(solutionProvider, xamlPullDiagnosticService)
+        { }
+
+        public override TextDocumentIdentifier? GetTextDocumentIdentifier(WorkspaceDocumentDiagnosticsParams request) => null;
+
+        protected override WorkspaceDiagnosticReport CreateReport(TextDocumentIdentifier? identifier, VSDiagnostic[]? diagnostics, string? resultId)
+            => new WorkspaceDiagnosticReport { TextDocument = identifier, Diagnostics = diagnostics, ResultId = resultId };
+
+        protected override ImmutableArray<Document> GetDocuments(RequestContext context)
+        {
+            using var _ = ArrayBuilder<Document>.GetInstance(out var result);
+
+            // Collect all the opened documents from solution. 
+            // In XamlLanguageService, we are only able to retrieve diagnostic information for opened documents. 
+            // So this is the same error experience we have now in full VS scenario.
+            var projects = context.Solution.GetXamlProjects();
+
+            foreach (var project in projects)
+            {
+                // For XamlProject, project.Documents only inlucdes opened documents, so just add them to the result.
+                result.AddRange(project.Documents);
```

Comment:
This is an implementation detail of how XamlProject are currently being populated, which may change in the future because we really want to track all XAML documents some day.

Reply:
So maybe remove the comments?

---
In reply to: [519964732](https://github.com/dotnet/roslyn/pull/49145#discussion_r519964732) [](ancestors = 519964732)

Reply:
Yeah, I don't think they add value here

---
In reply to: [519980058](https://github.com/dotnet/roslyn/pull/49145#discussion_r519980058) [](ancestors = 519980058,519964732)